### PR TITLE
[21176] Strip all passed values in settings area

### PR DESCRIPTION
**Pull request by basiszwo**
https://community.openproject.org/work_packages/21176

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://basiszwo-21176-space-character.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://basiszwo-21176-space-character.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: app/controllers/settings_controller.rb [hunk 1]
```diff
@@ -45,8 +45,12 @@ def edit
       end
 
       settings.each do |name, value|
-        # remove blank values in array settings
-        value.delete_if(&:blank?) if value.is_a?(Array)
+        if value.is_a?(Array)
+          # remove blank values in array settings
+          value.delete_if(&:blank?)
+        else
+          value = value.strip
```
Comment (NobodysNightmare) [inline app/controllers/settings_controller.rb:52]: I thought about doing this myself.
Do you think this is a safe operation?
I am not quite aware of all the settings we have (including plugins) and stripping whitespace on **all** of them sounds like a rough/risky operation.

Comment (basiszwo) [inline app/controllers/settings_controller.rb:52]: I cannot assume a reason why a trailing whitespace is required in any case. can you?

Comment (NobodysNightmare) [inline app/controllers/settings_controller.rb:52]: I could only craft one (e.g. "enter allowed characters here")...
I also don't _think_ it should pose a problem, but I am not sure. Plus I am not fond that this is a viable solution to the general problem, e.g. talking about the work package: I could have many other errors in the hostname and different problems in other fields that will still go unnoticed...

But maybe it is better to start with a solution like you are proposing and then further improve from there...

**Pull request merged**